### PR TITLE
Generate tuples for heterogeneous lists

### PR DESCRIPTION
**Pull request by jacereda**
This still depends on https://github.com/metacraft-labs/python-deduckt/pull/3

The submodule should be updated after that one is merged in order to completely pass the tests.

### CODE REVIEW

File: core.nim [hunk 1]
```diff
@@ -126,7 +126,7 @@ proc toType*(node: Node): Type =
     assert false
 
 proc noRec(label: string, t: Type): bool =
-  if len(label) > 0 and label == t.label:
+  if len(label) > 0 and not t.isNil and label == t.label:
```
Comment (alehander42) [inline core.nim]: oh, good

File: tests/python/heterogeneous_list.py [hunk 2]
```diff
@@ -0,0 +1,2 @@
+
+a = [1,True,'x']
```
Comment (alehander42) [inline tests/python/heterogeneous_list.py]: Yep

**Review by alehander92 — APPROVED** [review]

**Comment by alehander42** [comment]
I still think in some cases we'll have heterogenous lists which should be seq-s but with fixed types like variants, but this is harder to catch.

Thank you a lot! Are you using py2nim for something?

**Comment by jacereda** [comment]
Not yet, just evaluating nim as a possible next step for a medium-sized python code base.

**Pull request merged**